src/Models/Concerns/StoreAsFlatfile.php

```
@@ -38,9 +39,15 @@ public static function bootStoreAsFlatfile()
 
         $model = (new static);
 
+        $dbPath = Flatfile::getDatabasePath();
```

We do an exists check here:

https://github.com/ryanmitchell/statamic-stache-sqlite/blob/719c8aadda8cff8e60bc438b1c51e3862767576b/src/ServiceProvider.php#L21-L25

Which means it happens once per request instead of once per model... this was one of the minor performance improvements.

Are you seeing an error this is fixing?

---

Ah that helped me figure out what was going on.  That `$fs->put()` is silently failing during the cache build command because the cache directory didn't exist.  So the solution is to actually just ensure the directory exists before calling put.